### DISCUSSION TITLE
Invalid Hostname running msfconsole in WSL2

### DISCUSSION THREAD

**Discussion by wardellcastles** [rapid7/metasploit-framework#19001] [Q&A]
I get this error running msfconsole in any Linux distro on WSL2.  
/usr/share/metasploit-framework/lib/rex/proto/dns/static_hostnames.rb:84:in `add': **Invalid hostname**: **Red7en**. (ArgumentError)
Note: Red7en is the name of my computer.
I do not get this message running msfconsole on a Linux virtual machine, only in WSL2.

**Comment by CodeSailer** [comment]
Same here, I tried with different distros(kali and ubuntu) but I get the same error.

**Comment by mikaelhelguson** [comment]
`sudo nano /opt/metasploit-framework/embedded/framework/lib/rex/proto/dns/static_hostnames.rb  `
`CTRL+/` go to line 84
change 
`raise ::ArgumentError.new("Invalid hostname: #{hostname}")`
to
`#raise ::ArgumentError.new("Invalid hostname: #{hostname}")`
CTRL+X to exit and Y and ENTER to save and close

**Comment by meta40r** [comment]
Thanks @mikaelhelguson for the solution. While it works temporarily but someone needs to fix in repo as well for permanent fix.

**Comment by wardellcastles** [comment]
yes @mikaelhelguson thanks for the solution although it be an odd one.  Just comment out the error!

**Comment by AssassinUKG** [comment]
On kali install, the file to edit can be found on same line 84 (nano, Ctrl + /, line 84)

` sudo nano /usr/share/metasploit-framework/lib/rex/proto/dns/static_hostnames.rb`

**Comment by jmartin-tech** [comment]
Ref #19012

**Comment by giunatale** [comment]
this is actually a malformed `/etc/hosts` from WSL2, while they fix it, you can remove the /etc/hosts auto generation editing the `wsl.conf` or add at sysinit (e.g. using the systemd support in WSL2) - needs to run with root perms since it's editing the hosts file:
`sed -i -E "s/([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+)\s+([a-zA-Z0-9_-]+)\.\s+\2/\1$(echo -e '\t')\2/" /etc/hosts`

**Comment by bcoles** [comment]
This should be fixed in #19012.
